code/modules/mob/mob_transformation_simple.dm

```diff
@@ -22,39 +22,41 @@
 		to_chat(usr, span_danger("Cannot convert into a new_player mob type."))
 		return
 
-	var/mob/M
+	var/mob/desired_mob
 	if(isturf(location))
-		M = new new_type( location )
+		desired_mob = new new_type(location)
 	else
-		M = new new_type( src.loc )
+		desired_mob = new new_type(src.loc)
 
-	if(!M || !ismob(M))
+	if(!desired_mob || !ismob(desired_mob))
 		to_chat(usr, "Type path is not a mob (new_type = [new_type]) in change_mob_type(). Contact a coder.")
-		qdel(M)
+		qdel(desired_mob)
 		return
 
 	if( istext(new_name) )
-		M.name = new_name
-		M.real_name = new_name
+		desired_mob.name = new_name
+		desired_mob.real_name = new_name
 	else
-		M.name = src.name
-		M.real_name = src.real_name
-
-	if(has_dna() && M.has_dna())
-		var/mob/living/carbon/C = src
-		var/mob/living/carbon/D = M
-		C.dna.transfer_identity(D)
-		D.updateappearance(mutcolor_update=1, mutations_overlay_update=1)
-	else if(ishuman(M))
-		var/mob/living/carbon/human/H = M
-		client?.prefs.safe_transfer_prefs_to(H)
-		H.dna.update_dna_identity()
-
-	if(mind && isliving(M))
-		mind.transfer_to(M, 1) // second argument to force key move to new mob
+		desired_mob.name = src.name
+		desired_mob.real_name = src.real_name
+
+	if(has_dna() && desired_mob.has_dna())
+		var/mob/living/carbon/old_mob = src
+		var/mob/living/carbon/new_mob = desired_mob
+		old_mob.dna.transfer_identity(new_mob, transfer_species = FALSE)
+		new_mob.updateappearance(mutcolor_update=1, mutations_overlay_update=1)
+	else if(ishuman(desired_mob) && (!ismonkey(desired_mob)))
+		var/mob/living/carbon/old_human
```

Comment:
old human is empty here? aren't you transfering prefs to null?

Reply:
My mistake, this was the remnant of a previous attempted solution to the problem. The !ismonkey check should be sufficent.